### PR TITLE
fix: handle 502 after tx broadcast

### PR DESCRIPTION
**Pull request by kamleshmugdiya**
Handle 502 error after transaction broadcast

This commit adds error handling for the scenario where `sendUserToBitgoGShare` succeeds but `getTxRequest` fails with a 502 status code.
In such cases, it throws a specific error message indicating that the transaction was broadcasted successfully, but BitGo failed to update the transaction status.
This change ensures that more informative error message is provided when encountering bad gateway error during the transaction status retrieval.

Ticket: COIN-92

### CODE REVIEW

File: modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts [hunk 1]
```diff
@@ -886,7 +886,17 @@ export class EcdsaUtils extends baseTSSUtils<KeyShare> {
       SendShareType.SShare,
       userSShare
     );
-    return await getTxRequest(this.bitgo, this.wallet.id(), txRequestObj.txRequestId);
+    try {
+      return await getTxRequest(this.bitgo, this.wallet.id(), txRequestObj.txRequestId);
+    } catch (error) {
+      if (error.status === 502) {
+        throw new Error(
+          'Transaction broadcasted successfully, but BitGo failed to update the transaction status ' + error
```
Comment (Vijay-Jagannathan) [inline modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts:894]: We seem to be doing a get and not update

**Review by Vijay-Jagannathan — COMMENTED** [review]
Can we add a test to replicate it using existing tests ?

File: modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts (hunk 1 above)
Comment (gianchandania) [reply modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts:894]: +1

**Review by pengyuc-bitgo — DISMISSED** [review]

File: modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts (hunk 1 above)
Comment (kamleshmugdiya) [reply modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts]: fixed it.

Comment (kamleshmugdiya) [reply modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts]: have added couple of tests.

File: modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts [hunk 2]
```diff
@@ -629,8 +639,17 @@ export class EddsaUtils extends baseTSSUtils<KeyShare> {
     );
 
     await sendUserToBitgoGShare(this.bitgo, this.wallet.id(), txRequestId, userToBitGoGShare, apiVersion);
-
-    return await getTxRequest(this.bitgo, this.wallet.id(), txRequestId);
+    try {
+      return await getTxRequest(this.bitgo, this.wallet.id(), txRequestId);
+    } catch (error) {
+      if (error.status === 502) {
```
Comment (mullapudipruthvik) [inline modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts:645]: do we have to do the same for other status codes like 504 ?

File: modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts [hunk 3]
```diff
@@ -886,7 +886,17 @@ export class EcdsaUtils extends baseTSSUtils<KeyShare> {
       SendShareType.SShare,
       userSShare
     );
-    return await getTxRequest(this.bitgo, this.wallet.id(), txRequestObj.txRequestId);
+    try {
+      return await getTxRequest(this.bitgo, this.wallet.id(), txRequestObj.txRequestId);
+    } catch (error) {
+      if (error.status === 502) {
```
Comment (bhavidhingra) [inline modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts:892]: can we use a variable here instead of hard-coded numeric?

Comment (zahin-mohammad) [inline modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts:892]: How can we be sure that all 502 errors here have a succesfull broadcast? The error handling here does not seem verbose enough and potentially misleading. What about other error codes? Lets discuss more over slack or in the wallets team focus hour.

**Review by zahin-mohammad — CHANGES_REQUESTED** [review]

**Pull request closed without merging**